pkg/loop/plugin_relayer.go

```diff
@@ -0,0 +1,212 @@
+package loop
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	pb "github.com/smartcontractkit/chainlink-relay/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+// PluginRelayerName is the name for [PluginRelayer]/[NewGRPCPluginRelayer].
+const PluginRelayerName = "relayer"
+
+type PluginRelayer interface {
+	NewRelayer(ctx context.Context, config string, keystore Keystore) (Relayer, error)
+}
+
+func PluginRelayerHandshakeConfig() plugin.HandshakeConfig {
+	return plugin.HandshakeConfig{
+		ProtocolVersion:  0,
+		MagicCookieKey:   "magic-key-relayer-todo",
+		MagicCookieValue: "magic-value-relayer-todo",
```

Comment:
Will follow up on these soon here https://smartcontract-it.atlassian.net/browse/BCF-2198